homeassistant/components/wmspro/__init__.py

```diff
@@ -0,0 +1,57 @@
+"""The WMS WebControl pro API integration."""
+
+from __future__ import annotations
+
+import aiohttp
+from wmspro.webcontrol import WebControlPro
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN, MANUFACTURER
+
+PLATFORMS: list[Platform] = [Platform.COVER]
+
+type WebControlProConfigEntry = ConfigEntry[WebControlPro]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: WebControlProConfigEntry
+) -> bool:
+    """Set up wmspro from a config entry."""
+    host = entry.data[CONF_HOST]
+    session = async_get_clientsession(hass)
+    hub = WebControlPro(host, session)
+
+    try:
+        await hub.ping()
+        await hub.refresh()
+    except aiohttp.ClientError as err:
+        raise ConfigEntryNotReady(f"Error while connecting to {host}") from err
+
+    entry.runtime_data = hub
+
+    device_registry = dr.async_get(hass)
+    device_registry.async_get_or_create(
+        config_entry_id=entry.entry_id,
+        identifiers={(DOMAIN, hub.host)},
```

Comment:
we don't want to use hosts as identifiers. If there is nothing unique for the hub, use the `entry.entry_id`

Reply:
I know, but how can I get to the config_entry ID from within the entity, because I would then need to have it here also instead of the host:
https://github.com/home-assistant/core/blob/f9a2408dfa814f5e581b6a121b214a00a933b541/homeassistant/components/wmspro/generic_entity.py#L31

Reply:
just pass it as parameter?

Reply:
You mean store the config_entry in the entity on initial creation?

Reply:
you can just pass the entry_id and store that right?

Reply:
I solved it like this in the meantime, is that also okay?
https://github.com/mback2k/home-assistant/blob/8138541d453c34002e15f2698687a6519dde03e5/homeassistant/components/wmspro/generic_entity.py#L21-L35

Reply:
I don't really see a point in putting in the whole object only for one string

Reply:
Changed again.